src/editor-server.js

```diff
@@ -0,0 +1,160 @@
+import FalcorServer from 'falcor-express';
+import express from 'express';
+import React from 'react';
+import falcor from 'falcor';
+import _ from 'lodash';
+import { renderToString } from 'react-dom/server';
+import { match, RouterContext } from 'react-router';
+import sourcemap from 'source-map-support';
+import routes from 'lib/editor-routes';
+import FalcorController from 'lib/falcor/FalcorController';
+import FalcorRouter from 'lib/falcor/FalcorRouter';
+import { injectModelCreateElement } from 'lib/falcor/falcorUtils';
+
+// Load in static issue articles for development
+import articles from '../static/sample-issue/posts.js';
+import authors from '../static/sample-issue/authors.js';
+
+// Allow node to use sourcemaps
+sourcemap.install();
+
+
+const buildHtmlString = (body, cache) => {
+  return (
+    `<!DOCTYPE html>
+      <html style="height: 100%; padding: 0; margin: 0;">
+        <head>
+          <title>Hello World</title>
+        </head>
+        <body style="height: 100%; padding: 0; margin: 0;">
+          <div id="main" style="height: 100%; padding: 0; margin: 0;">`
+            + body +
+          `</div>
+          <script>
+            var _initialCache =
+            ` + JSON.stringify(cache) + `
+            ;
+          </script>
+          <script src="/build/editor-client.js"></script>
+        </body>
+      </html>`
+  );
+};
+
+// Shared serverModel
+// You can also hardcode / stub parts of the model here
+const serverModel = new falcor.Model({
+  /*cache: {
+    articlesBySlug:articles,
+    authorsBySlug:authors, 
+  },*/
+  source: new FalcorRouter()
+}).batch();
+
+// Asynchronously render this application
+// Returns a promise
+const renderApp = (renderProps) => {
+  let falcorPaths = _.compact(renderProps.routes.map((route) => {
+    const component = route.component;
+    if (component.prototype instanceof FalcorController) {
+      return component.getFalcorPath(renderProps.params);
+    }
+    return null;
+  }));
+  // Merging pathsets
+  let temp = [];
```

Comment:
Guess it's bad competitive programming habits :).

And yeah I'll use reduce. Then I won't need the temp variable either right? Isn't it better to use the built in to Javascript array method .reduce though? Instead of the library _.reduce? Or do you prefer _.reduce?